src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingVenueHelper.java

```diff
@@ -48,6 +50,48 @@ public static Map<String, String> getVenueToSearch(ListingData listingData) {
         }
     }
 
+    /**
+     * Returns Venue label from listing venue dynamic fixed list Item.
+     * This returns the dynamicFixedList labels for the hearingVenues
+     * with the ones which needs to be in report.
+     * @param listingData print hearing details from the case data
+     */
+    public static String getListingVenueLabel(ListingData listingData) {
+        Map<String, String> venueToSearchMap = getListingVenueLabelToSearch(listingData);
+        return venueToSearchMap.entrySet().iterator().next().getValue();
+    }
+
+    public static Map<String, String> getListingVenueLabelToSearch(ListingData listingData) {
```

Comment:
javadoc

Reply:
or...does this method need to be public?

Reply:
java doc added for this

Reply:
and similar to James, does this need to be public?

Reply:
There's a test written for this, private method can't be tested, catch moved to previous line

Reply:
removed test and made it private

Reply:
test was written separately for this method because illegalException can't be checked on the public calling method which sonar complains about